### DISCUSSION TITLE
Using existing secret and configmap for credentials and s3 bucket options

### DISCUSSION THREAD

**Discussion by DreamingRaven** [vmware-tanzu/velero#6747] [Community support Q&A]
Hey community,
in my bare-metal cluster I use rook-ceph to provide filesystem, block, and object storage.
Rook ceph generates a secret and a configmap which look like this respectively cumulatively providing the credentials needed to use the object storage. My question is, is there a good way to use both of these to automatically populate velero backup and snapshot location credentials and configuration?

My current thoughts are that I could possibly use helms lookup functionality to populate the CRDs myself although this would require a double install the first time (once to generate the storage then the next to lookup the credentials). Is there any better way to do this?

Example secret by rook-ceph:
```
apiVersion: v1
data:
  AWS_ACCESS_KEY_ID: ++++++++
  AWS_SECRET_ACCESS_KEY: ++++++++
kind: Secret
metadata:
  creationTimestamp: '2023-09-02T07:17:37Z'
  finalizers:
    - objectbucket.io/finalizer
  labels:
    bucket-provisioner: rook-ceph.ceph.rook.io-bucket
  name: velero-bucket
  namespace: velero
  ownerReferences:
    - apiVersion: objectbucket.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ObjectBucketClaim
      name: velero-bucket
      uid: d960f983-2f1f-4ff7-8306-f8fd08c01d5a
  resourceVersion: '8702081'
  uid: 25a5b313-5aac-4fad-a0b1-df012cf3c920
type: Opaque
```
example configmap by rook-ceph:
```
apiVersion: v1
data:
  BUCKET_HOST: rook-ceph-rgw-ceph-objectstore.rook-ceph.svc
  BUCKET_NAME: velero-7f201335-4f8d-4e04-90e3-eb037f9bebda
  BUCKET_PORT: '80'
  BUCKET_REGION: ''
  BUCKET_SUBREGION: ''
kind: ConfigMap
metadata:
  creationTimestamp: '2023-09-02T07:17:37Z'
  finalizers:
    - objectbucket.io/finalizer
  labels:
    bucket-provisioner: rook-ceph.ceph.rook.io-bucket
  name: velero-bucket
  namespace: velero
  ownerReferences:
    - apiVersion: objectbucket.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ObjectBucketClaim
      name: velero-bucket
      uid: d960f983-2f1f-4ff7-8306-f8fd08c01d5a
  resourceVersion: '8702082'
  uid: 35d670ac-e149-410d-ac48-c6e9cf96d0ba
```

**Comment by shubham-pampattiwar** [comment]
@DreamingRaven I believe you will just have to create the velero backupstoragelocation secret and configuration yourself by using the data from the rook-ceph configmap and secret, cannot think of any other way.
